src/TuxArchitectStores.js

```diff
@@ -62,34 +62,64 @@ ArchitectChain.prototype.itOutputs = function (output) {
+    storeOrStoresToWaitFor = inputOrStore;
   }
-  //return __registerId__ from storeToWaitFor
-  storeRegistrationId = storeToWaitFor.__registerId__;
-  //if store does not have __registerId__ than it has not been registered to the dispatcher so throw an error
-  invariant(storeRegistrationId, 'store is waiting for a store that has not been registered to any actions.');
-  return storeRegistrationId;
+  //return  __registerId__ or set of __registerId__'s from storeOrStoresToWaitFor
+  storeRegistrationIdorIds = this._returnOneOrMultipleRegisterIds(storeOrStoresToWaitFor);
+  return storeRegistrationIdorIds;
 };
 
+//_returnOneOrMultipleRegisterIds FUNCTION: internal function, used by _getRegisterId to return one or a set registerId's from a store or set of stores
+//@param storeOrStoresToWaitFor ARRAY or STRING: input string or array of strings representing all stores being waited on
+ArchitectChain.prototype._returnOneOrMultipleRegisterIds = function(storeOrStoresToWaitFor) {
+  var storeRegistrationId;
+  var noRegisteredIdError = 'store is waiting for a store that has not been registered to any actions.';
+  //if storeOrStoresToWaitFor is an array holding multiple stores
+  if (Array.isArray(storeOrStoresToWaitFor)) {
+    var storeRegistrationIds = [];
+    var storeOrStoresToWaitForLength = storeOrStoresToWaitFor.length;
+    //grab the __registerId__ from each individual store and return them as an array
+    for (var i = 0; i < storeOrStoresToWaitForLength; i++) {
+      storeRegistrationId = storeOrStoresToWaitFor[i].__registerId__;
+      //if store does not have __registerId__ then it has not been registered to the dispatcher so throw an error
+      invariant(storeRegistrationId, noRegisteredIdError);
+      storeRegistrationIds.push(storeRegistrationId);
+    }
+    return storeRegistrationIds;
+  } else {
+    //return __registerId__ from storeOrStoresToWaitFor
+    storeRegistrationId = storeOrStoresToWaitFor.__registerId__;
+    //if store does not have __registerId__ then it has not been registered to the dispatcher so throw an error
+    invariant(storeRegistrationId, noRegisteredIdError);
+    return storeRegistrationId;
+  }
+};
+
+
 //_registerOutputToArchitecture FUNCTION: internal function, registers an output STRING key to the architecture OBJECT with the value of storeToArchitect so that the storeToArchitect can later be looked up by passing the same output STRING to itNeeds
 //@param output STRING: output STRING to which storeToArchitect will be mapped to on the architecture OBJECT
 ArchitectChain.prototype._registerOutputToArchitecture = function (output) {
-  //if this output is already registered to architecture OBJECT, throw error
-  invariant(!architecture[output], 'output: "%s" is already registered to a store.', output);
-  //map output STRING to storeToArchitect
-  architecture[output] = this.storeToArchitect;
+  //if this output is already registered to architecture OBJECT
+  if (architecture[output]) {
```

Comment:
Store `architect[output]` in a variable like `mappedStoreOrStores`.  That way you can use it in the if statement and in the concat statement on line 118.  We still need to store the result in `architecture[output]` but that way we don't need to look up the result both in the if statement and to pass in to the `concat` method.